importer/src/bin/geojson_to_osmosis.rs

```diff
@@ -0,0 +1,61 @@
+use std::io::{self, Read};
+use geojson::{GeoJson, Value};
+
+/// Convert geojson boundary suitable for osmfilter and other osmosis based tools.
+/// Expects the input to no elements other than the boundary of interest.
+//
+/// Reads geojson text from stdin
+/// Writes "poly" formatted text to stdout
+fn main() -> Result<(), Error> {
+    let mut buffer = String::new();
+    io::stdin().read_to_string(&mut buffer)?;
+    let geojson = buffer.parse::<GeoJson>()?;
+    let points = boundary_coords(&geojson)?;
+
+    println!("map_name");
+    println!("1");
+    for point in points {
+        println!("     {}    {}", point[0], point[1]);
+    }
+    println!("END");
+    println!("END");
+    Ok(())
+}
+
+fn boundary_coords(geojson: &GeoJson) -> Result<Vec<Vec<f64>>, Error> {
+    let feature = match geojson {
+        GeoJson::Feature(feature) => feature,
+        GeoJson::FeatureCollection(feature_collection) => &feature_collection.features[0],
+        _ => return Err(Error::BadInput(format!("Unexpected geojson: {:?}", geojson)))
+    };
+
+    match &feature.geometry.as_ref().map(|g| &g.value) {
+        Some(Value::MultiPolygon(multi_polygon)) => {
+            return Ok(multi_polygon[0][0].clone())
+        },
+        Some(Value::Polygon(polygon)) => {
+            return Ok(polygon[0].clone())
+        },
+        _ => Err(Error::BadInput(format!("Unexpected feature: {:?}", feature)))
+    }
+}
+
+#[derive(Debug)]
```

Comment:
Since nothing tests for different error types later, a less boilerplate method is to return `Box<dyn std::error::Error>`, and then produce custom strings by returning `Err(format!("foo").into())`. The explicit enum is fine if that doesn't work or you prefer to build out a nicer API.

Reply:
fixed, I think. Please take a look to see if I captured what you were describing.

Reply:
Yup, LGTM! I've bounced back and forth on an error type (I think there was `abstutil::Error(String)` at some point), but then read this trick somewhere and found it the easiest for the "don't care" case

Reply:
I recently started using thiserror, which is super nice for libs where you want to expose proper errors, but agree that for top level binaries like this or something quick-n-dirty it's nice to avoid as much ceremony as possible.